File: 1_C/pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    // prompt user for input
    float dollars;
    int cents;
    do
    {
        dollars = get_float("change owed: ");
        // convert dollars to cents
        cents = round(dollars * 100);
    }
    while (dollars < 0);
    //printf("%i\n", cents);

    // define the coins available
    int money[4];
    money[0] = 25;
    money[1] = 10;
    money[2] = 5;
    money[3] = 1;

    //counting coins
    int coins = 0;
    for (int i = 0; i < 4; i++)
    {
        while (cents >= money[i])
        {
            cents -= money[i];
            coins++;
        }
    }
    printf("%i\n", coins);
}

File: 6_python/pset6/cash/cash.py
from cs50 import get_float


while True:
    n = get_float("Change owed: ")
    if (isinstance(n, float) == True) and (n > 0):
        cents = int(n * 100)
        break

# create list of coins
money = [25, 10, 5, 1]

# number of coins starts with 0
coins = 0
for i in money:
    while cents >= i:
        cents -= i
        coins += 1

print(coins)